js/outputter.js

```diff
@@ -0,0 +1,568 @@
+        this.bindSnippetEvents();
+    }
+};
+
+/**
+ * binds the analyzeTimer function to the input of the targetElement on the page.
+ */
+AnalyzeLoader.prototype.bindInputEvent = function() {
+    var elem = document.getElementById( this.config.elementTarget );
+    elem.__refObj = this;
+    elem.addEventListener( "input", this.analyzeTimer );
+};
+
+/**
+ * binds the reloadSnippetText function to the blur of the snippet inputs.
+ */
+AnalyzeLoader.prototype.bindSnippetEvents = function() {
+    var snippetElem = document.getElementById(this.config.targets.snippet);
+    snippetElem.refObj = this;
+    snippetElem.addEventListener("input", this.callBackSnippetData);
+    var elems = ["meta", "cite", "title"];
+    for (var i = 0; i < elems.length; i++) {
+        var targetElement = document.getElementById("snippet_" + elems[i]);
+        targetElement.refObj = this;
+        targetElement.addEventListener( "blur", this.reloadSnippetText );
+    }
+};
+
+/**
+ * callBackSnippetData is bound on the inputs of the snippetPreview, to update the inputs when entering text
+ * in the snippetPreview.
+ */
+AnalyzeLoader.prototype.callBackSnippetData = function() {
+    document.getElementById( "titleInput" ).value = document.getElementById( "snippet_title" ).innerText;
+    document.getElementById( "metaInput" ).value = document.getElementById( "snippet_meta" ).innerText;
+    document.getElementById( "urlInput" ).value = document.getElementById( "snippet_cite" ).innerText;
+};
+
+/**
+ * runs the rerender function of the snippetPreview if that object is defined.
+ */
+AnalyzeLoader.prototype.reloadSnippetText = function() {
+    if( typeof this.refObj.snippetPreview !== "undefined" ) {
+        this.refObj.snippetPreview.reRender();
+    }
+};
+
+/**
+ * the analyzeTimer calls the checkInputs function with a delay, so the function won' be executed at every keystroke
+ */
+AnalyzeLoader.prototype.analyzeTimer = function() {
+    var refObj = this.__refObj;
+    clearTimeout( window.timer );
+    window.timer = setTimeout( refObj.checkInputs, refObj.config.typeDelay );
+};
+
+/**
+ * calls the getInput function to retreive values from inputs. If the keyword is empty calls message, if keyword is filled, runs the analyzer
+ */
+AnalyzeLoader.prototype.checkInputs = function() {
+    var refObj = window.analyzeLoader;
+
+    refObj.getInput();
+    if( refObj.inputs.keyword === "" ){
+        refObj.showMessage();
+    }else{
+        refObj.runAnalyzer();
+    }
+
+};
+
+/**
+ * used when no keyword is filled in, it will display a message in the target element
+ */
+AnalyzeLoader.prototype.showMessage = function() {
+    this.target.innerHTML = "";
+    var messageDiv = document.createElement( "div" );
+    messageDiv.className = "wpseo_msg";
+    messageDiv.innerHTML = "<p><strong>No focus keyword was set for this page. If you do not set a focus keyword, no score can be calculated.</strong></p>";
```

Comment:
Will this be translatable when it is about to be implemented in for example WP SEO?